$ X = \left (\begin{array}{cccc}
1 & 0 \\
1 & 1 \\
\vdots& \vdots \\
1 & n
\end{array} \right) \cdot 
\left  (\begin{array}{cccc}
\beta_1 \\
\beta_2 
\end{array} \right) + 
\left (\begin{array}{cccc}
\varepsilon_0 \\
\varepsilon_0 + \varepsilon_1 \\
\vdots \\
\varepsilon_1 + \ldots + \varepsilon_n
\end{array} \right)$

$ Y = \left (\begin{array}{cccc}
x_0 \\
x_1 - x_0 \\
\vdots \\
x_n - x_{n-1}
\end{array} \right) = \left (\begin{array}{cccc}
\beta_1 + \varepsilon_0 \\
\beta_2 + \varepsilon_1 \\
\beta_1 + 2\cdot \beta_2 + \varepsilon_0 + \varepsilon_1 + \varepsilon_2
    -\beta_1 - \beta_2 - \varepsilon_0 - \varepsilon_1 \\
\vdots
\end{array} \right) =$

$ = \left(\begin{array}{cccc}
\beta_1 + \varepsilon_0 \\
\beta_2 + \varepsilon_1 \\
\beta_2 + \varepsilon_2 \\
\vdots \\
\beta_2 + \varepsilon_n
\end{array} \right) = \left(\begin{array}{cccc}
\beta_1 \\
\beta_2 \\
\vdots \\
\beta_2
\end{array} \right) + \left(\begin{array}{cccc}
\varepsilon_0 \\
\varepsilon_1 \\
\vdots \\
\varepsilon_n
\end{array} \right) = \left(\begin{array}{cccc}
1 & 0 \\
0 & 1 \\
0 & 1 \\
\vdots & \vdots \\
0 & 1
\end{array} \right) \cdot \left(\begin{array}{cccc}
\beta_1 \\
\beta_2
\end{array} \right) + \left(\begin{array}{cccc}
\varepsilon_0 \\
\vdots \\
\varepsilon_n
\end{array} \right)$

$\widehat{\theta} = (Z^T\cdot Z)^{-1} Z^T\cdot Y$

$Z^T\cdot Z = \left (\begin{array}{cccc}
1 & 0 \\
0 & n
\end{array} \right) \Rightarrow (Z^T\cdot Z)^{-1} =  \left(\begin{array}{cccc}
1 & 0 \\
0 & \frac{1}{n}
\end{array} \right)$

$(Z^T\cdot Z)^{-1}\cdot Z^T = \left (\begin{array}{cccc}
1 & 0 \\
0 & \frac{1}{n}
\end{array} \right) \cdot \left (\begin{array}{cccc}
1 & 0 & \ldots & 0 \\
0 & 1 & \ldots & 1
\end{array} \right) = \left(\begin{array}{cccc}
1 & 0 & \ldots & 0 \\
0 & \frac{1}{n} & \ldots & \frac{1}{n}
\end{array} \right)$

$(Z^T\cdot Z)^{-1}\cdot Z \cdot Y$ = $\left(\begin{array}{cccc}
1 & 0 & \ldots & 0 \\
0 & \frac{1}{n} & \ldots & \frac{1}{n}
\end{array} \right) \cdot \left(\begin{array}{cccc}
x_0 \\
\frac{1}{n} \cdot (x_1 - x_0 + x_2 - x_1 + \ldots)
\end{array} \right) = $

$//(x_1 - x_0) + (x_2 - x_1) + (x_3 - x_2) + \ldots + (x_n - x_{n - 1}) = x_n - x_0//$

$ = \left (\begin{array}{cccc}
x_0 \\
\frac{x_n - x_0}{n}
\end{array} \right)$ - оценка  $\left(\begin{array}{cccc}
\beta_1 \\
\beta_2
\end{array} \right)$

In [10]:
import numpy as nmp
import matplotlib.pyplot as plt
import matplotlib.mlab as mlab
import scipy.stats as sts
%matplotlib inline

# считываем данные
regr_file = []
file = open('Regression.csv', 'r')
for list in file:
    for value in list.split():
        regr_file.append(float(value))

size = len(regr_file) -1
# подставляем данные в полученную формулу        
estimate_beta = [regr_file[0], (regr_file[size] - regr_file[0]) / (size + 1)]
print(estimate_beta)

[104.9407, 13.9189515]


$\widehat{\sigma^2} = \frac{1}{(n+1) - k} \cdot E || Y - Z \cdot \widehat{\theta}||^2$ - 
несмещенная оценка квадрата дисперсии

$\sigma^2 = \frac{1}{n-1} \cdot E ||Y - Z\cdot \widehat{\theta}||^2$

$Z\cdot \widehat{\theta} = \left (\begin{array}{cccc}
1 & 0 \\
0 & 1 \\
\vdots & \vdots \\
0 & 1
\end{array} \right) \cdot \left (\begin{array}{cccc}
x_0 \\
\frac{x_n - x_0}{n}
\end{array} \right) = \left (\begin{array}{cccc}
x_0 \\
\frac{x_n - x_0}{n} \\
\vdots \\
\frac{x_n - x_0}{n}
\end{array} \right)$

$Y - Z\widehat{\theta} = \left (\begin{array}{cccc}
x_0 \\
x_1 - x_0 \\
\vdots \\
x_n - x_{n-1}
\end{array} \right) - \left (\begin{array}{cccc}
x_0 \\
\frac{x_n - x_0}{n} \\
\vdots \\
\frac{x_n - x_0}{n}
\end{array} \right) = \left (\begin{array}{cccc}
0 \\
x_1 - \frac{x_n}{n} - x_0\cdot \frac{n-1}{n} \\
x_2 - x_1 - \frac{x_n}{n} - \frac{x_0}{n} \\
\vdots \\
x_n - x_{n-1} - \frac{x_n}{n} + \frac{x_0}{n}
\end{array} \right)$

$||Y - Z \cdot \widehat{\theta}||^2 = $

$ = 0 + (x_1 - x_0 - \frac{x_n - x_0}{n})^2 + 
    \ldots + (x_n - x_{n-1} - \frac{x_n-x_0}{n})^2 = $
 
$ = (x_1 - x_0)^2 + \ldots + (x_n - x_{n-1})^2 - 
    2\cdot (x_1 - x_0)\cdot \frac{x_n - x_0}{n} - \ldots - 2\cdot (x_n - x_{n-1}) \cdot \frac{x_n - x_0}{n} +
    n\cdot \frac{(x_n - x_0)^2}{n} = $
    
$ = (x_1 - x_0)^2 + \ldots + (x_n - x_{n-1})^2 - \frac{(x_n - x_0)^2}{n}$
 
$\Rightarrow \widehat{\sigma^2} = \frac{1}{n-1}(\sum_{i=1}^n (x_i - x_{i-1}) - \frac{(x_n - x_0)}{n})^2$

$\Rightarrow \widehat{\sigma_t^2} = \frac{\widehat{\sigma^2}}{\widehat{\beta_2}^2}$

In [9]:
_sum = 0
# считаем сумму
for i in range(1, len(regr_file)):
    _sum = _sum + (regr_file[i] - regr_file[i - 1] - 
                   (regr_file[i] - regr_file[0]) / i)  ** 2

# подставляем значения в формулу
estimate_sigma = 1 / size * _sum
print(estimate_sigma)

# подставляем знаечения в следующую формулу
estimate_sigma_t  = estimate_sigma / (estimate_beta[1] ** 2)
print(estimate_sigma_t)

2.1961605300186466
0.011335770347251448


Вывод:

$\beta_1 = X_0 = 104.9407$

$\beta_2 = \frac{X_n - X_0}{n} = 13.9189515$

$\widehat{\sigma^2} = \frac{1}{n - 1} \cdot (\sum^n_{i=1} (X_i - X_{i - 1}) - \frac{(X_n - X_0)}{n})^2 = 2.1961605300186466$

$\widehat{\sigma^2_t} = \frac{\widehat{\sigma^2}}{\widehat{\beta_2}^2} = 0.011335770347251448$

Можно сделать вывод, что для данной выборки линейная модель подходит. Это можно объяснить тем, что движение трамва близко к равномерному. Соответственно, можно соответствующим образом оценить скорость $\beta_1$. $\beta_2$ - это оценка начального расстояния, и она совпадает с показанием датчика в нулевой момент времени.